ovn_k8s/modes/overlay.py

```diff
@@ -436,3 +466,366 @@ def sync_services(self, services):
+
+    def add_to_address_set(self, pod_ip, namespace, policy_data):
+        for rule in policy_data['spec']['ingress']:
+            rule_address_set = self._build_rule_address_set_name(
+                policy_data['metadata']['name'],
+                namespace,
+                rule)
+            ovn_nbctl('--if-exists', 'add', 'address_set',
+                      rule_address_set, 'addresses', pod_ip)
+
+    def remove_from_address_set(self, pod_ip, namespace, policy_data):
+        for rule in policy_data['spec']['ingress']:
+            rule_address_set = self._build_rule_address_set_name(
+                policy_data['metadata']['name'],
+                namespace,
+                rule)
+            ovn_nbctl('--if-exists', 'remove', 'address_set',
+                      rule_address_set, 'addresses', pod_ip)
+
+    def build_pseudo_acls(self, policy_data):
+        # Build pseudo ACL list for policy
+        policy_id = policy_data['metadata']['uid']
+        policy_pseudo_acls = []
+        for rule in policy_data['spec']['ingress']:
+            ports_data = rule.get('ports', [])
+            protocol_ports = []
+            for item in ports_data:
+                protocol_ports.append((item['protocol'], item.get('port')))
+            from_data = rule.get('from')
+            if not from_data:
+                src_pod_ips = '*'
+            else:
+                rule_address_set = self._build_rule_address_set_name(
+                    policy_data['metadata']['name'],
+                    policy_data['metadata']['namespace'],
+                    rule)
+                src_pod_ips = 'address_set(%s)' % rule_address_set
+            pseudo_acl = (POLICY_ACL_PRIORITY,
+                          protocol_ports, src_pod_ips,
+                          'allow-related')
+            policy_pseudo_acls.append(pseudo_acl)
+        self.pseudo_acls[policy_id] = policy_pseudo_acls
+        return policy_pseudo_acls
+
+    def remove_pseudo_acls(self, policy_id):
+        self.pseudo_acls.pop(policy_id, None)
+
+    def _find_policies_for_pod(self, pod_data, policies):
+        pod_labels = pod_data['metadata'].get('labels', {})
+        pod_policies = []
+        for policy in policies:
+            policy_spec = policy['spec']
+            pod_selector = policy_spec.get('podSelector', {}).get(
+                'matchLabels', {})
+            # TODO(salv-orlando): Implement not only equality based selectors
+            if pod_selector:
+                for label in set(pod_labels.keys()) & set(pod_selector.keys()):
+                    if pod_labels[label] == pod_selector[label]:
+                        pod_policies.append(policy)
+                        # policy matched, move on
+                        break
+            else:
+                # An empty selector matches all pods in the namespace
+                pod_policies.append(policy)
+        return pod_policies
+
+    def apply_pod_policy_acls(self, pod_data, policies):
```

Comment:
I will add proper docstrings for all the functions which are supposed to be part of the public interface, and simple comments for the others